test/test_transformers.py

```diff
@@ -1880,16 +1913,33 @@ def test_mem_efficient_attention_vs_math_ref_grads(self, device, batch_size: int
                                                                                        is_causal=is_causal, scale=scale))
                 return
             else:
-                out = F.scaled_dot_product_attention(
-                    query, key, value, dropout_p=dropout_p, is_causal=is_causal, scale=scale)
+                # Set the seed and run the kernel
+                torch.manual_seed(seed)
+                with sdp_kernel(enable_mem_efficient=True, enable_flash=False, enable_math=False):
+                    out = F.scaled_dot_product_attention(
+                        query, key, value, dropout_p=dropout_p, is_causal=is_causal, scale=scale)
 
-        with sdp_kernel(enable_math=True, enable_flash=False, enable_mem_efficient=False):
+        if dropout_p == 0.0:
+            with sdp_kernel(enable_math=True, enable_flash=False, enable_mem_efficient=False):
+                # High Precision Math Reference
+                out_ref = F.scaled_dot_product_attention(query_ref, key_ref, value_ref,
+                                                         dropout_p=dropout_p, is_causal=is_causal, scale=scale)
+                # Low Precision Math Reference
+                out_lp_ref = F.scaled_dot_product_attention(query_ref_lp, key_ref_lp, value_ref_lp,
+                                                            dropout_p=dropout_p, is_causal=is_causal, scale=scale)
+        else:
+            if seq_len_q > 1024:
+                self.skipTest("Will call _fill_mem_eff_dropout_mask with too many threads!")
+            # Create the dropout_mask
+            torch.manual_seed(seed)
+            dropout_mask = _get_mem_eff_drop_mask(batch_size, n_heads, seq_len_q, seq_len_k, dropout_p, seed, 0, device=device)
             # High Precision Math Reference
-            out_ref = F.scaled_dot_product_attention(query_ref, key_ref, value_ref,
```

Comment:
Why this change instead of running it under the guard with math set to True only?

Reply:
the dropout_mask arg isn't exposed at the top level so call the math func directly